### PR TITLE
[Relations] Fix xToOne displaying duplicate

### PR DESCRIPTION
**Pull request by ronronscelestes**
## Issue

**CONTENT-547**

`xToOne` relations field would sometimes display 2 relations:
- The saved one relation
- The new relation to replace the old one before saving

It was caused by the fact that we always display saved relations except if they exist in the disconnect array

## What

Refactor `CONNECT_RELATION` dispatch to properly handle `xToOne` / `replace`

## Test

- Create `xToOne` + `xToMany` relations
- `xToMany` should still allow multiple relations 
- `xToOne` should always display only 1 relation

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js [hunk 1]
```diff
@@ -95,27 +95,50 @@ const reducer = (state, action) =>
         const { value, replace = false } = action;
         const connectedRelations = get(state, [...path, 'connect']);
         const disconnectedRelations = get(state, [...path, 'disconnect']);
+        const savedRelations = get(state, [...path, 'results']);
```
Comment (joshuaellis) [inline packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js:98]: you're checking the existence of `disconnectRelations` and `savedRelations` a lot, you could do:

```ts
const savedRelations = get(state, [...path, 'results']) ?? [];
```

to save you some time and continuously checking if the array exists.

**Review by joshuaellis — APPROVED** [review]
Love to see the comments 🤩

File: packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js (hunk 1 above)
Comment (ronronscelestes) [reply packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js:98]: I added it and I was able to remove the check on savedRelations in
```js
const existInSavedRelation =
          savedRelations?.findIndex((savedRelations) => savedRelations.id === value.id) !== -1;
``` 

most of the time I check on length though, didn't find another place where I could remove this type of check - did I miss something? 👀

Comment (joshuaellis) [reply packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js]: CleanShot 2022-10-11 at 13 29 24@2x

all your `ifs` are `disconnectedRelations?.length` (essentially checking if the length is above 0) so by ensuring `disconnectedRelations` is _definitely_ an `array` via a `??` you can just just "disconnectedRelations.length` which is easier to understand and reduces the amount of optional chaining we do.

**Comment by ronronscelestes** [comment]
@joshuaellis should be ready!
don't hesitate to fully test `xToOne` `xToMany` relations as the refactor updated all `CONNECT_RELATION` dispatch - works on my side but I hope I didn't miss anything 💥

**Review by joshuaellis — APPROVED** [review]
Tested on my side and works 👌🏼

**Pull request merged**